src/service/url-replacements-impl.js

```diff
@@ -163,6 +164,14 @@ export class UrlReplacements {
       return removeFragment(info.sourceUrl);
     }));
 
+    this.setAsync_('SOURCE_URL', () => {
+      return getTrackImpressionPromise().then(
+        this.getDocInfoValue_.bind(this, info => {
```

Comment:
my questions are:
1. Not only `SOURCE_URL` needs the `info.sourceUrl`,  `SOURCE_HOST` `SOURCE_HOSTNAME`...all needs that info. When I update the source url, should I make a copy and keep the old value? (cuz `SOURCE_HOST` `SOURCE_HOSTNAME` may still needs the old values?)
2. If we only update source url only for the case `SOURCE_URL` I don't see why it's necessary to add this update method to `document-info`. why don't we just call `this.ampdoc.win.location.href` since no one will ever use that again. 
3. It's interesting that we never to `document-info` provided url anywhere in `url-replaces-impl`, we always get from `win.location.href` directly. Why is that?